lib/roo/excelx/sheet_doc.rb

```diff
@@ -82,34 +82,41 @@ def cell_value_type(type, format)
       #
       # Returns a type of <Excelx::Cell>.
       def cell_from_xml(cell_xml, hyperlink)
-        coordinate = extract_coordinate(cell_xml['r'])
-        return Excelx::Cell::Empty.new(coordinate) if cell_xml.children.empty?
+        coordinate = ::Roo::Utils.extract_coordinate(cell_xml['r'.freeze])
+        cell_xml_children = cell_xml.children
+        return Excelx::Cell::Empty.new(coordinate) if cell_xml_children.empty?
 
         # NOTE: This is error prone, to_i will silently turn a nil into a 0.
         #       This works by coincidence because Format[0] is General.
-        style = cell_xml['s'].to_i
-        format = styles.style_format(style)
-        value_type = cell_value_type(cell_xml['t'], format)
+        style = cell_xml['s'.freeze].to_i
         formula = nil
 
-        cell_xml.children.each do |cell|
+        cell_xml_children.each do |cell|
           case cell.name
-          when 'is'
-            content_arr = cell.search('t').map(&:content)
-            unless content_arr.empty?
-              return Excelx::Cell.create_cell(:string, content_arr.join(''), formula, style, hyperlink, coordinate)
+          when 'is'.freeze
+            content = ""
+            cell.children.each do |inline_str|
+              if inline_str.name == 't'.freeze
+                content << inline_str.content
+              end
             end
-          when 'f'
+            unless content.empty?
+              return Excelx::Cell.cell_class(:string).new(content, formula, style, hyperlink, coordinate)
+            end
+          when 'f'.freeze
             formula = cell.content
-          when 'v'
-            return create_cell_from_value(value_type, cell, formula, format, style, hyperlink, base_date, coordinate)
+          when 'v'.freeze
+            format = style_format(style)
```

Comment:
Could we cache `format` and `value_type` for reuse on subsequent iterations? Considering they operate on the parent `cell_xml` and `style` there appears to be reuse.

Reply:
No we don't need to cache format and value_type as we use return statement in next line